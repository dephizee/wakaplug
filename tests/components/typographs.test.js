import React from 'react';
import renderer from 'react-test-renderer';
import { AuthHeaderStyled, AuthLabelStyled, AuthSpanStyled, AuthSubHeaderStyled, AuthTextStyled, LinkStyled, PasswordToggle, SpanStyled } from "./../../components/typographs.style";

test('Header working', () => {
  const component = renderer.create(
    <AuthHeaderStyled>Facebook</AuthHeaderStyled>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  // tree.props.onMouseEnter();
  // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // manually trigger the callback
  // tree.props.onMouseLeave();
  // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});

test('Sub Header working', () => {
  const component = renderer.create(
    <AuthSubHeaderStyled>Twitter</AuthSubHeaderStyled>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  
});

test('AuthText working', () => {
  const component = renderer.create(
    <AuthTextStyled>LinkedIn</AuthTextStyled>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('AuthLabel working', () => {
  const component = renderer.create(
    <AuthLabelStyled>2go</AuthLabelStyled>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LinkStyled working', () => {
  const component = renderer.create(
    <SpanStyled>Youtube</SpanStyled>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LinkStyled working', () => {
  const component = renderer.create(
    <LinkStyled>Tinder</LinkStyled>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('AuthSpan working', () => {
  const component = renderer.create(
    <AuthSpanStyled>Uber</AuthSpanStyled>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('PasswordToggle working', () => {
  const component = renderer.create(
    <PasswordToggle>Bolt</PasswordToggle>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});