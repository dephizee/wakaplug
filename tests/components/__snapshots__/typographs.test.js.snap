// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLabel working 1`] = `
<label
  className="sc-hKwDye WjGTI"
>
  2go
</label>
`;

exports[`AuthSpan working 1`] = `
<span
  className="sc-iCfMLu heaqjd"
>
  Uber
</span>
`;

exports[`AuthText working 1`] = `
<p
  className="sc-dkPtRN ceZDnr"
>
  LinkedIn
</p>
`;

exports[`Header working 1`] = `
<h4
  className="sc-bdvvtL fNwMSt"
>
  Facebook
</h4>
`;

exports[`LinkStyled working 1`] = `
<span
  className="sc-eCImPb hxoRsn"
>
  Youtube
</span>
`;

exports[`LinkStyled working 2`] = `
<a
  className="sc-gKclnd iqcaJE"
>
  Tinder
</a>
`;

exports[`PasswordToggle working 1`] = `
<span
  className="sc-jRQBWg fTKDDY"
>
  Bolt
</span>
`;

exports[`Sub Header working 1`] = `
<h5
  className="sc-gsDKAQ jGIkdn"
>
  Twitter
</h5>
`;
